isobus/include/isobus/isobus/isobus_device_descriptor_object_pool.hpp

```diff
@@ -0,0 +1,142 @@
+		bool add_device_process_data(std::string processDataDesignator,
+		                             std::uint16_t processDataDDI,
+		                             std::uint16_t deviceValuePresentationObjectID,
+		                             std::uint8_t processDataProperties,
+		                             std::uint8_t processDataTriggerMethods,
+		                             std::uint16_t uniqueID);
+
+		/// @brief Adds a device property object to the DDOP
+		/// @param[in] propertyDesignator Descriptive text for the object, UTF-8, 32-128 chars max
+		/// @param[in] propertyValue The value of the property
+		/// @param[in] propertyDDI Identifier of property (DDI) according to definitions in Annex B and ISO 11783 - 11.
+		/// @param[in] valuePresentationObject Object identifier of DeviceValuePresentationObject, or NULL object ID
+		/// @param[in] uniqueID The object ID of the object. Must be unique in the DDOP.
+		/// @returns `true` if the object was added to the DDOP, `false` if the object cannot be added (duplicate or some other error)
+		bool add_device_property(std::string propertyDesignator,
+		                         std::int32_t propertyValue,
+		                         std::uint16_t propertyDDI,
+		                         std::uint16_t valuePresentationObject,
+		                         std::uint16_t uniqueID);
+
+		/// @brief Adds a device value presentation object to the DDOP
+		/// @param[in] unitDesignator Unit designator for this value presentation
+		/// @param[in] offsetValue Offset to be applied to the value for presentation.
+		/// @param[in] scaleFactor Scale to be applied to the value for presentation.
+		/// @param[in] numberDecimals Specifies the number of decimals to display after the decimal point.
+		/// @param[in] uniqueID The object ID of the object. Must be unique in the DDOP.
+		/// @returns `true` if the object was added to the DDOP, `false` if the object cannot be added (duplicate or some other error)
+		bool add_device_value_presentation(std::string unitDesignator,
+		                                   std::int32_t offsetValue,
+		                                   float scaleFactor,
+		                                   std::uint8_t numberDecimals,
+		                                   std::uint16_t uniqueID);
+
+		/// Constructs a binary DDOP using the objects that were previously added
+		/// @param[in,out] resultantPool The binary representation of the DDOP, or an empty vector if this function returns false
+		/// @returns `true` if the object pool was generated and is valid, otherwise `false`.
+		bool generate_binary_object_pool(std::vector<std::uint8_t> &resultantPool);
+
+		/// @brief Gets an object from the DDOP that corresponds to a certain object ID
+		/// @returns Pointer to the object matching the provided ID, or nullptr if no match was found
+		task_controller_object::Object *get_object_by_id(std::uint16_t objectID);
```

Comment:
Might be safer to return an `weak_ptr` or `shared_ptr`? Might be worth it just to be sure we never end up with dangling pointers, even though it would require refactoring objectList to use `shared_ptr` instead. What do you think?

Reply:
Seems like a smart idea, I can clean this up